File: 2021/csharp/Aoc2021.Tests/Day01Tests.cs
namespace Aoc2021.Tests;

public class Day01Tests
{
    [Fact]
    public void CountIncreasedSeaDepths()
    {
        int[] seaDepths = { 199, 200, 208, 210, 200, 207, 240, 269, 260, 263 };
        Assert.Equal(7, Day01.CountIncreasedSeaDepths(seaDepths));
    }

    [Fact]
    public void CountIncreasedSeaDepthWindows()
    {
        int[] seaDepths = { 199, 200, 208, 210, 200, 207, 240, 269, 260, 263 };
        Assert.Equal(5, Day01.CountIncreasedSeaDepthWindows(seaDepths));
    }

    [Fact]
    public void Stars()
    {
        Day01 run = new();
        Assert.Equal("1583", run.Result1());
        Assert.Equal("1627", run.Result2());
    }
}

File: 2021/fsharp/Aoc2021.Tests/Day01.Tests.fs
module Day01.Tests

open Xunit
open Utils
open Day01

let seaDepths =
    "199 200 208 210 200 207 240 269 260 263" |> String.parseInts ' '

[<Fact>]
let ``Day01 count increased sea depths works`` () =
    Assert.Equal(7, countIncreasedSeaDepths seaDepths)

[<Fact>]
let ``Day01 count increased sea depth windows works`` () =
    Assert.Equal(5, countIncreasedSeaDepthWindows seaDepths)

[<Fact>]
let ``Day01 Stars`` () =
    Assert.Equal("1583", job1 ())
    Assert.Equal("1627", job2 ())
